tests/unit/output/test_local.py

```diff
@@ -1,25 +1,52 @@
-from mock import patch
+import os
 
+from mock import patch
 from dvc.output import OutputLOCAL
 from dvc.remote.local import RemoteLOCAL
 from dvc.stage import Stage
+from dvc.utils import relpath
 from tests.basic_env import TestDvc
 
 
-class TestOutputLOCAL(TestDvc):
+class OutputLOCALBase(TestDvc):
     def _get_cls(self):
         return OutputLOCAL
 
-    def _get_output(self):
-        stage = Stage(self.dvc)
-        return self._get_cls()(stage, "path", cache=False)
+    def _get_output(self, wdir=None, path="path"):
+        stage = Stage(self.dvc, wdir=wdir if wdir else os.curdir)
+        return self._get_cls()(stage, path, cache=False)
+
 
+class TestOutputLOCAL(OutputLOCALBase):
     def test_save_missing(self):
         o = self._get_output()
         with patch.object(o.remote, "exists", return_value=False):
             with self.assertRaises(o.DoesNotExistError):
                 o.save()
 
+    def test_str_if_workdir_is_inside_repo(self):
+        work_dir = os.path.join(self.root_dir, "models")
+        os.mkdir(work_dir)
+        os.chdir(work_dir)
```

Comment:
With `os.chdir()` you are changing global state within a test and don't restore it in the end.